First Implementation

livghit opened the pull request:
## 📑 Description
First implementation of the Goravel installer 

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

## ℹ Additional Information
It's not perfect , still needs some improvements .

hwbrzzl commented on go.mod:
```diff
@@ -0,0 +1,21 @@
+module github.com/goravel/installer
+
+go 1.22.1
```
```suggestion
go 1.21
```

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+var rootCmd = &cobra.Command{
+	Use:   "goravel",
+	Short: "Goravel installer",
+	Long:  `This is the goravel installer , inspired by Laravel , build with love and care`,
```
```suggestion
	Long:  `This is the goravel installer, build with love and care`,
```

hwbrzzl commented on ui/styles.go:
```diff
@@ -0,0 +1,38 @@
+
+	LogoStyle = lipgloss.NewStyle().
+			Bold(true).
+			Foreground(errorColor)
```
The color can be #78D5FB(logo).

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+func generate(projectName string) {
+	fmt.Println(ui.DefaultMessage.Render("This clones the repo and so on inside ", projectName))
+	os := runtime.GOOS
+	switch os {
+	case "darwin":
+		generateForUnix(projectName)
+		return
+	case "windows":
+		generateForWindows(projectName)
+		return
```
Should support `linux`.

livghit replied:
Linux would be the same as dawin the commands are the same if I don't  missthink right now . I'll add the case for it too .

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+		generateForWindows(projectName)
+		return
+	default:
+		fmt.Println(ui.ErrorMessage.Render("Unsupported platform. Closing"))
```
Need to list the supported platform, and the current platform here.

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+	}
+	fmt.Println(ui.DefaultMessage.Render("Repo cloned"))
+
+	remove_git := exec.Command("rm", "-rf", projectName+"/.git", projectName+"/.github")
```
```suggestion
	removeFiles := exec.Command("rm", "-rf", projectName+"/.git", projectName+"/.github")
```

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+	fmt.Println(ui.DefaultMessage.Render("Instaling the Goravel"))
+	install := exec.Command("go", "mod", "tidy")
+	install.Dir = ("./" + projectName)
+	install.Run()
```
Should to handle the error.

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+	install.Dir = ("./" + projectName)
+	install.Run()
+
+	fmt.Print(ui.SuccessMessage.Render("Goravel installed sucessfuly !"))
```
```suggestion
	fmt.Print(ui.SuccessMessage.Render("Goravel installed sucessfully!"))
```

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+
+	fmt.Print(ui.SuccessMessage.Render("Goravel installed sucessfuly !"))
+
+	cp_env := exec.Command("cp", ".env.example", ".env")
```
```suggestion
	copeEnv := exec.Command("cp", ".env.example", ".env")
```

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+
+	fmt.Println(ui.DefaultMessage.Render("Generating app key "))
+
+	app_key := exec.Command("go", "run", ".", "artisan", "key:", "generate")
```
```suggestion
	initAppKey := exec.Command("go", "run", ".", "artisan", "key:", "generate")
```

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+	}
+	fmt.Println(ui.DefaultMessage.Render("Repo cloned"))
+
+	remove_git := exec.Command("Remove-Item", "-Path", "./"+projectName+"/.git", "./"+projectName+"/.github", "-Recursive", "-Force")
```
```suggestion
	removeFiles := exec.Command("Remove-Item", "-Path", "./"+projectName+"/.git", "./"+projectName+"/.github", "-Recursive", "-Force")
```

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+	}
+
+	fmt.Println(ui.DefaultMessage.Render("Generating app key "))
+	cp_env := exec.Command("cp", ".env.example", ".env")
```
```suggestion
	copyEnv := exec.Command("cp", ".env.example", ".env")
```

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+	fmt.Print(ui.SuccessMessage.Render("Goravel installed sucessfuly !"))
+	fmt.Println(ui.DefaultMessage.Render("Generating app key "))
+
+	app_key := exec.Command("go", "run", ".", "artisan", "key:", "generate")
```
```suggestion
	initAppKey := exec.Command("go", "run", ".", "artisan", "key:", "generate")
```

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,163 @@
+		panic(err)
+	}
+
+	fmt.Println("You can cd into your project and start developing ")
```
We can append the command here, like: `cd ** && go run .`

livghit commented:
Hey ,

Thanks for the review , I'll restyle the UI and implement the sugested comments . Thank you :)

livghit commented:
I added all the wanted changes beside the UI. I'll need a little bit more time for it ... :D

hwbrzzl commented:
> I added all the wanted changes beside the UI and cd ** && go run . I'll need a little bit more time for it ... :D

We only need to append the words to print, like: `fmt.Println("You can cd into your project and start developing: cd ** && go run .")`

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,169 @@
+
+	removeFiles := exec.Command("rm", "-rf", projectName+"/.git", projectName+"/.github")
+
+	err = removeFiles.Run()
+	if err != nil {
```
```suggestion
	if err := removeFiles.Run(); err != nil {
```

hwbrzzl commented on cmd/main.go:
```diff
@@ -0,0 +1,169 @@
+
+	err = removeFiles.Run()
+	if err != nil {
+		panic(err)
```
Can we print a kind error message to control instead of panic?

livghit replied:
> Can we print a kind error message to control instead of panic?

Do you mean removing the panic and only logging the error ? Thank you for all the help :D

hwbrzzl replied:
Yes, panic is ugly.

hwbrzzl reviewed (commented):
Could you please attach the latest UI image here?

livghit commented:
> Could you please attach the latest UI image here?

<img width="765" alt="image" src="https://github.com/goravel/installer/assets/108449432/77da68b9-20a9-4452-929b-6092070a343e">

I tried to get also the output of cloning the repo and so on but run into errors. I may need some help with that .

hwbrzzl commented:
The UI looks better! 👍  Please pin me when it's ready to be reviewed again.

livghit commented:
@hwbrzzl Well I think it is kinda ready . I tried to improve the UI as much as I could , I added error handling with log.Fatalf and fixed some typos . 

The UI of the new command looks like this :
<img width="606" alt="image" src="https://github.com/goravel/installer/assets/108449432/cd18597b-927d-4b97-8926-9f37d14f14d9">

The UI looks like this when an error happens : 
<img width="1215" alt="image" src="https://github.com/goravel/installer/assets/108449432/be3c357a-1cea-449d-857a-98be5e39d510">

I am aware that there may be a lot to improve but at the moment I am not sure if I am capable to do it in a short time.

Thank you very much !

hwbrzzl reviewed (approved):
Yes, although some optimizations can be made, but it's good enough for now. I'll add you to the contributor list, thanks so much for your hard work!
